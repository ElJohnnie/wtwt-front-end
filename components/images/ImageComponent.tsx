import React from 'react';
import Image from 'next/image';
import { ImageComponentProps } from '../../types';

export default function ImageComponent(props: ImageComponentProps) {
  const { href, title, description } = props;
  return (
    <div className="flex flex-col justify-center items-center mx-auto">
      <figure className="w-full">
        <Image
          layout="responsive"
          src={href}
          alt="image description"
          width={80}
          height={80}
        />
      </figure>
      <h3 className="font-bold text-slate-900 text-center text-lg mt-3 max-w-x">
        {title}
      </h3>
      <p className="text-slate-700 text-center text-sm mt-2 max-w-sm">
        {description}
      </p>
    </div>
  );
}
