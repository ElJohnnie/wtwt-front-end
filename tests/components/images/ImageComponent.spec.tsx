import React from 'react';
import ImageComponent from '../../../components/images/image.component';
import { renderWithOutContextProvider } from '../../../utils/tests/helpers';

describe('ImageComponent', () => {
  it('quero que minha imagem renderize sem problemas', () => {
    const { container } = renderWithOutContextProvider(
      <ImageComponent
        href="https://image.tmdb.org/t/p/w500/pB8BM7pdSp6B6Ih7QZ4DrQ3PmJK.jpg"
        title="Clube da Luta"
        description="Um homem deprimido que sofre de insônia conhece um estranho vendedor chamado Tyler Durden e se vê morando em uma casa suja depois que seu perfeito apartamento é destruído. A dupla forma um clube com regras rígidas onde homens lutam. A parceria perfeita é comprometida quando uma mulher, Marla, atrai a atenção de Tyler."
        rating={4}
      />,
    );
    expect(container).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
