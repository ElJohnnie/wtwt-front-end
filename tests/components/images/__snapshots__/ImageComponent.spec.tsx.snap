// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageComponent quero que minha imagem renderize sem problemas 1`] = `
<div>
  <div
    class="flex flex-col justify-center items-center mx-auto"
  >
    <figure
      class="w-full"
    >
      <img
        alt="image description"
        data-nimg="1"
        decoding="async"
        height="80"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw500%2FpB8BM7pdSp6B6Ih7QZ4DrQ3PmJK.jpg&w=3840&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw500%2FpB8BM7pdSp6B6Ih7QZ4DrQ3PmJK.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw500%2FpB8BM7pdSp6B6Ih7QZ4DrQ3PmJK.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw500%2FpB8BM7pdSp6B6Ih7QZ4DrQ3PmJK.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw500%2FpB8BM7pdSp6B6Ih7QZ4DrQ3PmJK.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw500%2FpB8BM7pdSp6B6Ih7QZ4DrQ3PmJK.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw500%2FpB8BM7pdSp6B6Ih7QZ4DrQ3PmJK.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw500%2FpB8BM7pdSp6B6Ih7QZ4DrQ3PmJK.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw500%2FpB8BM7pdSp6B6Ih7QZ4DrQ3PmJK.jpg&w=3840&q=75 3840w"
        style="color: transparent; width: 100%; height: auto;"
        width="80"
      />
    </figure>
    <h3
      class="font-bold text-pink-600 text-center text-lg mt-3 max-w-x"
    >
      Clube da Luta
    </h3>
    <p
      class="text-pink-600 text-center text-sm mt-2 max-w-sm"
    >
      Um homem deprimido que sofre de insônia conhece um estranho vendedor chamado Tyler Durden e se vê morando em uma casa suja depois que seu perfeito apartamento é destruído. A dupla forma um clube com regras rígidas onde homens lutam. A parceria perfeita é comprometida quando uma mulher, Marla, atrai a atenção de Tyler.
    </p>
  </div>
</div>
`;
